src/plugins/offline/azureOfflinePlugin.ts

```diff
@@ -60,14 +58,17 @@ export class AzureOfflinePlugin extends AzureBasePlugin {
   }
 
   private async azureOfflineBuild(){
-    await this.offlineService.build();
+    const offlineService = new OfflineService(this.serverless, this.options);
```

Comment:
probably not too much of a difference, but what are we given up creating a new service everytime?

Reply:
We can't initialize a `BaseService` within the constructor of a plugin. The constructor of `BaseService` assumes that the core framework has made whatever config substitutions it needs to do

Reply:
This looks good enough for now - but I feel the `BaseService` is starting to do too much and we should consider scaling it back at some point.  Ex) It does a lot of config initialization and if you think about all the different services we instantiate in the course of calling a plugin the same code is getting run over and over.